modules/core/src/main/scala/dev/profunktor/redis4cats/effect/RedisEc.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018-2020 ProfunKtor
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.profunktor.redis4cats.effect
+
+import cats.effect._
+import java.util.concurrent.Executors
+
+trait RedisEc {
+  def delay[F[_]: Sync: ContextShift, A](thunk: => A): F[A]
+  def eval[F[_]: ContextShift, A](fa: F[A]): F[A]
+}
+
+object RedisEc {
+  def apply(blocker: Blocker): RedisEc =
+    new RedisEc {
+      override def delay[F[_]: Sync: ContextShift, A](thunk: => A): F[A] = blocker.delay(thunk)
+      override def eval[F[_]: ContextShift, A](fa: F[A]): F[A]           = blocker.blockOn(fa)
```

Comment:
I think there should be a single method, maybe just `eval`? `blocker.blockOn` should not be used, there are no blocking operations AFAIK.

Reply:
Oh nevermind, seems correct. `blockOn` is a misleading name but I guess it means to evaluate that `fa` in the given EC and shift back, right?